airflow/models/taskinstance.py

```diff
@@ -299,11 +299,14 @@ def clear_task_instances(
         )
         dag_run_state = DagRunState(dag_run_state)  # Validate the state value.
         for dr in drs:
-            dr.state = dag_run_state
-            dr.start_date = timezone.utcnow()
-            if dag_run_state == DagRunState.QUEUED:
-                dr.last_scheduling_decision = None
-                dr.start_date = None
+            # Otherwise task(s) cleared in the middle of the running DagRun.
+            # There is no need to reset anything
+            if dr.state in State.finished_dr_states:
```

Comment:
The comment says in the middle of running, but the if clause says _in a finished state_. They don’t seem to match?

Reply:
I can delete that comment if you want :) I just wanted to highlight that we don't want to reset state in the middle of running dag run

Reply:
It’d make more sense if the comment says something like _only reset state if DAG run has finished_ instead (i.e. match the actual condition).